main.tf

```diff
@@ -15,6 +15,10 @@
  */
 
 locals {
+
+  // project could be the same as var.project_id or different in case jenkins instance creating in shared VPC (means network belongs to differnet project)
+  jenkins_network_project_id = regex("/projects/([^/]*)/", var.jenkins_instance_network)[0]
```

Comment:
We can't assume that `var.jenkins_instance_network` is a self link including a host project ID; it may only be a network ID. We need to retrieve the host project ID with a data source.

Reply:
@aaron-lane  data source for what resource

Reply:
A [google_compute_network data source](https://www.terraform.io/docs/providers/google/d/datasource_compute_network.html) should be able to provide the project like `data.google_compute_network.main.project`.

Reply:
but when we'll define data "google_compute_network" "main" we'll need to provide project id otherwise it will use provider default, means we need to provider project id initially, can't retrieve it

Reply:
Oh, good point. Let's an an optional `jenkins_network_project_id` variable with a default of `null`. Then this local can be adjusted like: `jenkins_network_project_id = var.jenkins_network_project_id == null ? var.project_id : var.jenkins_network_project_id`.

Reply:
got it, it was my first option, but I switched to current one as more elegant, so coming back to initial idea